### PR TITLE
add OSGi compatibility

### PR DESCRIPTION
**Pull request by maggu2810**
Let's create a manifest, so that we can use the Chromecast API in an OSGi environment (e.g. in some IoT context).

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
     <groupId>su.litvak.chromecast</groupId>
     <artifactId>api-v2</artifactId>
     <version>0.9.2-SNAPSHOT</version>
-    <packaging>jar</packaging>
+    <packaging>bundle</packaging>
```
Comment (vitalidze) [inline pom.xml]: Why do you change it? I don't plan to put a 'bundle' to the Maven Central.

File: pom.xml [hunk 2]
```diff
@@ -21,8 +21,8 @@
     <scm>
         <url>https://github.com/vitalidze/chromecast-java-api-v2</url>
         <connection>scm:git:https://github.com/vitalidze/chromecast-java-api-v2.git</connection>
-      <tag>HEAD</tag>
-  </scm>
+        <tag>HEAD</tag>
```
Comment (vitalidze) [inline pom.xml]: Please leave original indentation.

**Comment by vitalidze** [comment]
Generally, it's ok. But I don't like changing of 'packaging'. Maybe it's worth to implement this as a submodule.

File: pom.xml (hunk 2 above)
Comment (maggu2810) [inline pom.xml]: I am fine with you that a PR that adds OSGi compatibility should not change the indentation of some other lines.

Just one comment :wink: 

You are using an indentation step with of four spaces and the whole pom.xml is well formatted / indent except this two lines.
The tag "connection" and "tag" are sibling childs of scm and on the same level.
connection is using eight spaces, tag is using four spaces.
The open tag "scm" is using four spaces, the closing one two.

I have assumed that this is not done by purpose and want give you a consistent indentation of the whole XML.

I can revert this.

File: pom.xml (hunk 1 above)
Comment (maggu2810) [inline pom.xml]: Do you know that the artifact is still the "same" of type jar but with a OSGi compatible manifest?
We do not change the artifact type here, only grant that the maven-bundle-plugin of Apache Felix is used for packaging (and it could add a manifest by inspection of the classes).

We could also leave the packaging type as it is.
See "Adding OSGi metadata to existing projects without changing the packaging type" here "http://felix.apache.org/documentation/subprojects/apache-felix-maven-bundle-plugin-bnd.html"

But could you (please) explain me your concerns as long as the resulting artifact type is not changed?

Comment (vitalidze) [inline pom.xml]: My main concern is that I don't know what the 'bundle' means here and it's not a standard packaging time for maven project (see http://maven.apache.org/pom.html#Maven_Coordinates). It confuses me and I don't want to spend time investigating why it is put there and what it will mean. Neither I want the library users to spend their time for doing same investigation. I believe that this is a quite more specific thing than the 'jar' packaging. I think that this parameter is one of the major parameter of the maven projects.

 So, if you want it to be merged then please change it to the scheme where it updates the manifest as said in that link.

File: pom.xml (hunk 2 above)
Comment (vitalidze) [inline pom.xml]: Please revert this. You are welcome to submit a separate pull request with indentation changes if needed.

File: pom.xml (hunk 1 above)
Comment (maggu2810) [inline pom.xml]: > So, if you want it to be merged then please change it to the scheme where it updates the manifest as said in that link.

Sure, will do so, soon. It is your project.

> Neither I want the library users to spend their time for doing same investigation.

The user do not need to spent time to investigate this.
I assume you do also spent no time to investigate why the protobuf of google is using bundle packaging :wink: 
http://search.maven.org/#artifactdetails%7Ccom.google.protobuf%7Cprotobuf-java%7C2.6.0%7Cbundle

Will update this PR.

Thank you for your time.

File: pom.xml (hunk 2 above)
Comment (maggu2810) [inline pom.xml]: Will do so.

**Comment by maggu2810** [comment]
done

**Comment by vitalidze** [comment]
Merged. Thank you.
